engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/text/RegexCompileNode.java

```diff
@@ -0,0 +1,69 @@
+package org.enso.interpreter.node.expression.builtin.text;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.exception.AbstractTruffleException;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.source.Source;
+
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.builtin.Builtins;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(
+    type = "Prim_Text_Helper",
+    name = "compile_regex",
+    description = "Compiles a regexp.",
+    autoRegister = false)
+public abstract class RegexCompileNode extends Node {
+  static RegexCompileNode build() {
+    return RegexCompileNodeGen.create();
+  }
+
+  abstract Object execute(Object self, Object pattern, Object options);
```

Comment:
```suggestion
  abstract Object execute(Object pattern, Object options);
```

If I see correctly, `compile_regex` is static. So that means no `self`.